### PR TITLE
Feature/logging

### PR DESCRIPTION
**Pull request by shatteredbeam**
New logging implementation.

### CODE REVIEW

**Comment by shatteredbeam** [comment]
Prefixes have been removed as requested.  The module name is now included in the main log string.

File: Modules/rat_board.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ class RescueBoardException(BaseException):
     """
 
     def __init__(self, *args: object) -> None:
-        LOG.error(f"Rescueboard Exception raised!\nargs = {args}")
+        log.error(f"RBoard:  Rescueboard Exception raised!\nargs = {args}")
```
Comment (theunkn0wn1) [inline Modules/rat_board.py]: Ahem

File: Modules/rat_board.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ def __contains__(self, other: Rescue) -> bool:
             raise TypeError
         else:
             for rescue in self.rescues.values():
-                LOG.debug(f"checking rescue {rescue} against {other}...\n"
+                log.debug(f"RBoard:  Checking rescue {rescue} against {other}...\n"
```
Comment (theunkn0wn1) [inline Modules/rat_board.py]: ahem

File: Modules/rat_board.py [hunk 3]
```diff
@@ -257,19 +257,19 @@ def append(self, rescue: Rescue, overwrite: bool = False) -> None:
             result = False
         # check if its equal to what we already have
         elif found == rescue:
-            LOG.debug("a call was made to modify, yet the rescue was not changed!")
+            log.debug("RBoard:  A call was made to modify, yet the rescue was not changed!")
```
Comment (theunkn0wn1) [inline Modules/rat_board.py]: .

File: Modules/rat_board.py [hunk 4]
```diff
@@ -257,19 +257,19 @@ def append(self, rescue: Rescue, overwrite: bool = False) -> None:
...
             raise RescueNotChangedException
         else:
             # its not what we already have
 
             # lets check if we have a API handler
             if self.handler is not None:
-                LOG.debug("Rescue has been modified, making a call to the API")
+                log.debug("RBoard:  Rescue has been modified, making a call to the API")
```
Comment (theunkn0wn1) [inline Modules/rat_board.py]: .

File: Modules/rat_board.py [hunk 5]
```diff
@@ -257,19 +257,19 @@ def append(self, rescue: Rescue, overwrite: bool = False) -> None:
...
                 # if so, let it know we changed the rescue
                 # FIXME: change to match API Handler interface, once it exists
                 await self.handler.update_rescue(rescue)
 
-            LOG.debug(f"Updating local rescue #{rescue.board_index} (@{rescue.case_id}...")
+            log.debug(f"RBoard:  Updating local rescue #{rescue.board_index} (@{rescue.case_id}...")
```
Comment (theunkn0wn1) [inline Modules/rat_board.py]: .

File: Modules/rat_board.py [hunk 6]
```diff
@@ -305,7 +306,7 @@ def clear_board(self) -> None:
 
         useful for flushing the board prior to re-retrieving cases from the API
         """
-        LOG.warning("Flushing the Dispatch Board, fire in the hole!")
+        log.warning("RBoard:  Flushing the Dispatch Board, fire in the hole!")
```
Comment (theunkn0wn1) [inline Modules/rat_board.py]: .

File: Modules/rat_rescue.py [hunk 7]
```diff
@@ -689,7 +689,7 @@ def mark_for_deletion(self, value) -> None:
                     raise ValueError("Data validation failed. at least one key contains invalid"
                                      "data.")
             else:
-                LOG.debug(f"data of value is: {value}")
+                log.debug(f"Rescue:  data of value is: {value}")
```
Comment (theunkn0wn1) [inline Modules/rat_rescue.py]: .

File: Modules/rat_rescue.py [hunk 8]
```diff
@@ -764,7 +764,7 @@ def rats(self, value):
             elif found[1]:
                 # a generic match (not platform specific) was found
                 # TODO throw a warning so the invoking method can handle this condition
-                LOG.warning("A match was found, but it was not the right platform!")
+                log.warning("Rescue:  A match was found, but it was not the right platform!")
```
Comment (theunkn0wn1) [inline Modules/rat_rescue.py]: .

File: Modules/rats.py [hunk 9]
```diff
@@ -102,9 +102,9 @@ def uuid(self, value) -> None:
             None
         """
         if isinstance(value, str):
-            LOG.debug(f"value was a string with data '{value}'")
+            log.debug(f"Rats:  value was a string with data '{value}'")
```
Comment (theunkn0wn1) [inline Modules/rats.py]: .

File: Modules/rats.py [hunk 10]
```diff
@@ -102,9 +102,9 @@ def uuid(self, value) -> None:
...
             uuid = UUID(value)
-            LOG.debug("parsed value into a valid UUID.")
+            log.debug("Rats:  parsed value into a valid UUID.")
```
Comment (theunkn0wn1) [inline Modules/rats.py]: .

File: main.py [hunk 11]
```diff
@@ -36,9 +36,10 @@ class MechaClient(Client):
         Called upon connection to the IRC server
         :return:
         """
-        log.debug("on connect invoked")
+        log.debug(f"Bot:  Connecting to channels...")
```
Comment (theunkn0wn1) [inline main.py]: .

**Pull request merged**